platforms/native/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/NativeBinaryFixture.groovy

```diff
@@ -121,8 +124,19 @@ class NativeBinaryFixture {
         } else {
             def symbols = binaryInfo.listDebugSymbols()
             def symbolNames = symbols.collect { it.name }
+            def isOlderSwiftc = toolChain.meets(ToolChainRequirement.SWIFTC) && toolChain.version < VersionNumber.version(5, 10)
             sourceFileNames.each { sourceFileName ->
-                assert sourceFileName in symbolNames
+                if (sourceFileName in symbolNames) {
+                    return
+                }
+                if (isOlderSwiftc) {
```

Comment:
Swiftc seems to now add the source file names to the debugging information, so look for this first and only look for the object files if the source file names are not present.